### PR TITLE
add IndicSentimentClassification dataset for Indic languages

### PR DESCRIPTION
**Pull request by jaygala24**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by jaygala24** [comment]
Adding IndicSentiment - a multilingual n-way parallel dataset for sentiment analysis in Indic languages in the Classification task category.

Team:
- Diganta Mista (@digantamisra98) - MILA
- Jay Gala (@jaygala24) - AI4Bharat

File: mteb/tasks/Classification/multilingual/IndicSentimentClassification.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            self.dataset[lang] = datasets.load_dataset(
+                name=f"translation-{lang}",
+                **self.metadata_dict["dataset"],
+            )
+            self.dataset[lang] = datasets.DatasetDict(
+                {
+                    "train": self.dataset[lang]["validation"],
+                    "test": self.dataset[lang]["test"],
+                }
+            )
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/IndicSentimentClassification.py:77]: Could this potentially be omitted?

**Review by isaac-chung — COMMENTED** [review]
Great work! Just a small comment.
Don't forget the points :D

File: mteb/tasks/Classification/multilingual/IndicSentimentClassification.py (hunk 1 above)
Comment (jaygala24) [reply mteb/tasks/Classification/multilingual/IndicSentimentClassification.py:77]: `AbsTaskClassification` expects `train` split so I am using `validation` set as the `train` split in this case.

**Comment by isaac-chung** [comment]
The results are part of the requirements of the PR (which I missed the 1st round!), good catch! In terms of points, we should check whether the languages introduced in this PR are already covered in other classification datasets.
- This dataset = 2 pts
- Any new language not covered previously in classification = 4 pts

File: mteb/tasks/Classification/multilingual/IndicSentimentClassification.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+_LANGUAGES = {
+    "as": ["asm-Beng"],
+    "bd": ["brx-Deva"],
+    "bn": ["ben-Beng"],
+    "gu": ["guj-Gujr"],
+    "hi": ["hin-Deva"],
+    "kn": ["kan-Knda"],
+    "ml": ["mal-Mlym"],
+    "mr": ["mar-Deva"],
+    "or": ["ory-Orya"],
```
Comment (isaac-chung) [inline mteb/tasks/Classification/multilingual/IndicSentimentClassification.py:20]: I found this as `ori` instead: https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes
```suggestion
    "or": ["ori-Orya"],
```

**Comment by isaac-chung** [comment]
These languages are new in classification: asm, brx, mar, ori, pan = 5x4 = 20
Total would be 22 pts. Up to you on how to split amongst yourselves.

File: mteb/tasks/Classification/multilingual/IndicSentimentClassification.py (hunk 2 above)
Comment (jaygala24) [reply mteb/tasks/Classification/multilingual/IndicSentimentClassification.py:20]: `iso_639_3_to_language.json` in the repository uses `ory` as the language code for Odia.

https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/iso_639_3_to_language.json#L5089

**Comment by jaygala24** [comment]
> These languages are new in classification: asm, brx, mar, ori, pan = 5x4 = 20 Total would be 22 pts. Up to you on how to split amongst yourselves.

I have added the points for the dataset contribution and PR review.

**Pull request merged**